### PR TITLE
Second NanoAOD update

### PR DESCRIPTION
**Pull request by arizzi**
This PR contains:
- update of Gen information (dressed leptons, gen jet flavour etc..)
- fix bug Uint8 -> bool
- muon ID from muon POG object functions
- updated list of btag discriminators
- review of substructures information
- protect against non-thread safeness of StringParser
- update electron variables
- fix of L1 seed pt
- update of tau trig object
- addition of LHE NLO info

@gpetruc @peruzzim 

This PR can now be merged (of course once signed) [we decided to keep two separate branches for 94X and 10XY and we will take care of fwd porting from 94X into 10XY based branch]

### CODE REVIEW

**Comment by gpetruc** [comment]
@cmsbuild please test

**Comment by arizzi** [comment]
hold the tests, I cherry-picked the wrong commit

File: PhysicsTools/NanoAOD/plugins/GenJetFlavourTableProducer.cc [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    unsigned int ncand = 0;
+    std::vector<int> partonFlavour;
+    std::vector<uint8_t> hadronFlavour;
+    
+    for (const reco::GenJet & jet : *jets) {
+      if (!cut_(jet)) continue;
+      ++ncand;
+      bool matched = false;
+      for (const reco::JetFlavourInfoMatching & jetFlavourInfoMatching : *jetFlavourInfos) {
+        if (deltaR(jet.p4(), jetFlavourInfoMatching.first->p4()) < 0.1) {
```
[the remark this answers is not in the document]
Comment (arizzi) [reply PhysicsTools/NanoAOD/plugins/GenJetFlavourTableProducer.cc]: fixed

**Comment by gpetruc** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
+1
- Visual inspection of the code is ok
- Updates do not touch miniAOD or reco products, only nanoAOD ones

**Comment by davidlange6** [comment]
merge

**Pull request merged**